### PR TITLE
Return pointer to View func

### LINKED ISSUE #3199: May be better to return a pointer to View, since it is a pretty large struct
https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/metric/view/view.go#L54, also change methods to be on the pointer instead of value, to avoid unnecessary copy every time match is called.

### PR DESCRIPTION
**Pull request by jyz0309**
Signed-off-by: jyz0309 <45495947@qq.com>

Resolves #3199

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: sdk/metric/view/view.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ type View struct {
 // the match Options passed for the View to be applied to it. Similarly, all
 // transform operation Options are applied to matched Instruments.
 func New(opts ...Option) (View, error) {
```
Comment (MrAlias) [inline sdk/metric/view/view.go:51]:
```suggestion
func New(opts ...Option) (*View, error) {
```

**Review by MrAlias — CHANGES_REQUESTED** [review]
Please add a changelog entry for this.

**Comment by jmacd** [comment]
I'd like to see which benchmarks improve with this PR before and after, since the motivation for this change is based on performance.

**Comment by jyz0309** [comment]
> I'd like to see which benchmarks improve with this PR before and after, since the motivation for this change is based on performance.

Ok, I will make a benchmark for it this weekend. : )

**Comment by jyz0309** [comment]
> I'd like to see which benchmarks improve with this PR before and after, since the motivation for this change is based on performance.

I have made a small benchmark for View.New() function.
```
BenchmarkViewPointer/View.New
BenchmarkViewPointer/View.New-8         	1000000000	         0.0000240 ns/op
BenchmarkViewPointer/View.New_without_pointer
BenchmarkViewPointer/View.New_without_pointer-8         	1000000000	         0.0000252 ns/op
``` 
It seems that the benefit is very small :( I will close this PR later.

**Pull request closed without merging**